descartes_light/include/descartes_light/ladder_graph.h

```diff
@@ -76,17 +105,25 @@ class LadderGraph
    */
   void resize(const std::size_t n_rungs);
 
-  Rung& getRung(const std::size_t index) noexcept;
-  const Rung& getRung(const std::size_t index) const noexcept;
+  std::vector<Rung<FloatType>>& getRungs() noexcept;
+  const std::vector<Rung<FloatType>>& getRungs() const noexcept;
+
+  Rung<FloatType>& getRung(const std::size_t index) noexcept;
+  const Rung<FloatType>& getRung(const std::size_t index) const noexcept;
 
-  std::vector<EdgeList>& getEdges(const std::size_t index) noexcept;  // see p.23 Effective C++ (Scott Meyers)
-  const std::vector<EdgeList>& getEdges(const std::size_t index) const noexcept;
+  NodeList& getNodes(const std::size_t index) noexcept;
+  const NodeList& getNodes(const std::size_t index) const noexcept;
+
+  Node<FloatType>& getNode(const std::size_t rung, const std::size_t index);
+  const Node<FloatType>& getNode(const std::size_t rung, const std::size_t index) const;
+
+  EdgeList& getEdges(const std::size_t rung,
+                     const std::size_t index) noexcept;  // see p.23 Effective C++ (Scott Meyers)
+  const EdgeList& getEdges(const std::size_t rung, const std::size_t index) const noexcept;
 
   std::size_t rungSize(const std::size_t index) const noexcept;
```

Comment:
Do you think we really need all of these helper functions? If we're making const and non-const versions of them, it seems like we should just expose the `std::vector<Rung>` member and get rid of all these functions

Reply:
I will remove most of them which are needed and used.